### PR TITLE
homework 5 done

### PR DESCRIPTION
**Pull request by PIDAMI**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            })
+            .collect(Collectors.toList());
+        button.get(0).click();
+    }
+
+    public void clickButtonInService(String buttonText) {
+        List<WebElement> buttons = serviceButtons.stream()
+                                                 .filter(el -> el.getText().trim().equalsIgnoreCase(buttonText))
+                                                 .collect(Collectors.toList());
+        buttons.get(0).click();
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java]: Check button list is not empty before. It cause outOfBound exception.

File: src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @Step("Getting navigation buttons' text")
+    public void clickServiceButton(String buttonText) {
+        List<WebElement> button = navigationButtons
+            .stream()
+            .filter(el -> {
+                WebElement childElementWithText = el.findElement(By.xpath("./a"));
+                return childElementWithText.getText().trim().equalsIgnoreCase(buttonText);
+            })
+            .collect(Collectors.toList());
+        button.get(0).click();
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java]: Check button list is not empty before. It cause outOfBound exception.

File: src/test/java/com/epam/tc/hw5/entities/UserTableRow.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        UserTableRow other = (UserTableRow) o;
+
+        return (other.number == null || number == null || other.number.equals(number))
+            && (other.selectedValue == null || selectedValue == null || other.selectedValue.equals(selectedValue))
+            && (other.description == null || description == null || other.description.equals(description))
+            && (other.userName == null || userName == null || other.userName.equals(userName))
+            && (other.isCheckboxChecked == null || isCheckboxChecked == null
+                || other.isCheckboxChecked.equals(isCheckboxChecked));
+    }
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw5/entities/UserTableRow.java]: Can it be simplified? Looks a bit complicated to understand and debug in future.

**Review by Delnote — COMMENTED** [review]
1. Remove all unused methods, fields and other stuff if exist.
2. Try to simplify locators especially with list of classes

File: src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java (hunk 1 above)
Comment (PIDAMI) [reply src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java]: added list size check

File: src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java (hunk 2 above)
Comment (PIDAMI) [reply src/test/java/com/epam/tc/hw5/page/objects/composite/components/HeaderComponent.java]: added list size check

File: src/test/java/com/epam/tc/hw5/entities/UserTableRow.java (hunk 3 above)
Comment (PIDAMI) [reply src/test/java/com/epam/tc/hw5/entities/UserTableRow.java]: refactored with use of reflection: now I get all declared fields and check for each one if this object's value's null or equal to other's value

**Comment by Delnote** [comment]
Done

**Pull request merged**